### PR TITLE
Announce WildFly cloud new architecture

### PR DESCRIPTION
**Pull request by jfdenise**
@jmesnil , early PR for the new WildFly Images to get feedbacks. Thank-you. Should be published on monday.

### CODE REVIEW

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+* New WildFly S2I builder images (link:https://quay.io/repository/wildfly/wildfly-s2i-jdk11[JDK11] and link:https://quay.io/repository/wildfly/wildfly-s2i-jdk17[JDK17])
+* New WildFly runtime images (link:https://quay.io/repository/wildfly/wildfly-runtime-jdk11[JDK11] and link:https://quay.io/repository/wildfly/wildfly-runtime-jdk17[JDK17])
+* New link:https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin 3.0.0.Final] 
+* New link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[WildFly cloud Galleon feature-pack 1.0.0.Final] 
+* New link:https://github.com/wildfly/wildfly-charts[Helm Chart for WildFly 2.0] 
+
+The new S2I workfow can be summerized as:
+
+* Configure the WildFly Maven plugin to deploy your application in the WildFly server of your choice, finely tuned according to your needs (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/jsf-ejb-jpa/pom.xml[an example])!
+* Make your application sources available in a git repository (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/postgresql[an example]).
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:25]: s/git/Git

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+* Define a Helm Chart for WildFly yaml file (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/postgresql/helm.yaml[an example]).
+* Install your helm file into OpenShift (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/jsf-ejb-jpa/README.md#example-steps[an example]). You are done!
+
+To get you familiar with the various components you can directly jump to these link:https://github.com/wildfly/wildfly-s2i/tree/main/examples[examples]. 
+They will drive you through various use-cases from source to deployment on OpenShift (for immediate access to an OpenShift cluster, you can use link:https://developers.redhat.com/developer-sandbox[OpenShift sandbox]).
+
+== Legacy WildFly S2I and runtime centos7 images
+
+We have released the legacy images for WildFly 26.1.0 in link:https://quay.io/organization/wildfly[quay.io/wildfly].
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:34]: either "WildFly 26.1" or "WildFly 26.1.0.Final" would be more appropriate

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+Our new images are sharing the Java VM configuration link:https://github.com/jboss-container-images/openjdk/blob/develop/modules/jvm/api/module.yaml[API] with the link:https://github.com/jboss-container-images/openjdk[openjdk images]. 
+This has multiple benefits:
+
+* JDK alignement. Benefit from latest updates.
+* JVM configuration alignment. Benefit from the same automatic tuning and configuration of Java VM.
+
+In the end you are getting an homogeneous user experience whatever the kind of WildFly server packaging you have chosen: 
+WildFly Bootable JAR that runs on the openjdk images or WildFly that runs on our new images.
+
+=== An S2I builder image un-coupled from WildFly server
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:82]: s/un-coupled/decoupled ?

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+S2I builder images are generic and can be used to deploy application for any WildFly server version.
+
+Builder images have no dependency on a WildFly server release (as opposed to the legacy centos7 WildFly images that are bound to a given server version). 
+
+You can even run WildFly EE8 or WildFly EE9 based application on the same images.
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:88]: mmmh, "Jakarta EE8 or Jakarta EE9-based application from these images" ?

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+The link:https://github.com/wildfly/wildfly-s2i/tree/main/examples/postgresql/README.md[PostgreSQL example] and 
+link:https://github.com/wildfly/wildfly-s2i/tree/main/examples/postgresql-jakarta/README.md[PostgreSQL Jakarta EE9 example] 
+are identical applications deployed on different kind of WildFly server. Just a matter of choosing the right Galleon feature-packs to assemble the right server.
+
+=== Release of the new images
+
+Images release cadence is now un-related to the WildFly server release cadence. Images have their own lifecycle. New releases will be done to address fixes and new features.
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:95]: s/un-related/decoupled

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+[NOTE]
+====
+When using the JDK17 image, you need to set the following env variable to workaround link:https://issues.redhat.com/browse/WFCORE-5835[WFCORE-5835]:
+```
+JAVA_OPTS_APPEND=--add-exports=jdk.naming.dns/com.sun.jndi.dns=ALL-UNNAMED
+```
+====
+
+=== Multi arch images preview
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:113]: I think this should be written as "Multiarch"

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+* link:https://quay.io/repository/wildfly-snapshots/wildfly-s2i-jdk17-multi-arch[https://quay.io/repository/wildfly-snapshots/wildfly-s2i-jdk17-multi-arch] preview multi arch JDK17 S2I builder image.
+
+* link:https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk11-multi-arch[https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk11-multi-arch] preview JDK11 multi arch runtime image.
+
+* link:https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk17-multi-arch[https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk17-multi-arch] preview JDK17 multi arch runtime image.
+
+== New s2i build workflow
+
+In order to create a server to be installed into the WildFly image we are relaying on the
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:130]: s/relaying/relying

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+* link:https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk11-multi-arch[https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk11-multi-arch] preview JDK11 multi arch runtime image.
+
+* link:https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk17-multi-arch[https://quay.io/repository/wildfly-snapshots/wildfly-runtime-jdk17-multi-arch] preview JDK17 multi arch runtime image.
+
+== New s2i build workflow
+
+In order to create a server to be installed into the WildFly image we are relaying on the 
+link:https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin 3.0.0.Final] that can now provision a fully configured server containing your deployment. 
+The WildFly Maven plugin `3.0.0.Final` has been evolved with some new goals to provision, configure, and package the server and the deployment in one step. 
+When designing your application pom file, add the WildFly Maven plugin `package` goal, configure it with the link:https://docs.wildfly.org/26.1/Galleon_Guide.html#wildfly-galleon-feature-packs[WildFly Galleon feature-pack]
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:133]: s/add the WildFly Maven plugin `package` goal/add the WildFly Maven plugin *to* the `package` goal

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+and link:https://docs.wildfly.org/26.1/Galleon_Guide.html#wildfly_layers[Galleon layer(s)], and optionally reference WildFly CLI scripts to be executed and 
+content to be copied inside the server. At the end of the build you will get (by default in the `target/server` directory) 
+a server with your app deployed, ready to be installed in the image.
+
+In order to allow for a smooth transition to the new images, we are still supporting (in a deprecated way) the legacy workflow. 
+Your existing application would work, but you are now required to specify the Galleon feature-pack(s) and layer(s) 
+(`GALLEON_PROVISION_FEATURE_PACKS` and `GALLEON_PROVISION_LAYERS` env variables) 
+you want to see used during the S2I build to provision a WildFly server.
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:141]: you want to provision during the S2I build.

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+=== WildFly server subsystems configuration API
+
+If you are using WildFly s2i images you are perhaps asking yourself where are the env variables you have been using 
+to configure the elytron subsystem, to add datasources, to configure logging or the microprofile-config subsystem,... 
+They are provided by means of a new link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[WildFly cloud Galleon feature-pack] 
+that you can combine with the WildFly Galleon feature-pack at build time to produce a server supporting these env variables.
+
+* If you only provision `org.wildfly:wildfly-galleon-pack:26.1.0.Final` you will get a "vanilla" WildFly server that 
+will get lightly adjusted by the image entry-point to properly execute on OpenShift.
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:174]: s/entry-point/entrypoint

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+
+* If you provision `org.wildfly:wildfly-galleon-pack:26.1.0.Final` and `org.wildfly.cloud:wildfly-cloud-galleon-pack:1.0.0.Final` 
+you will get a WildFly server link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack/blob/main/doc/layers.md[pre configured] for cloud execution ready to be configured thanks to 
+the link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack/blob/main/doc/launch.md[environment variables] the cloud feature-pack exposes.
+
+2 variants exist of the cloud feature-pack:
+
+* `org.wildfly.cloud:wildfly-cloud-galleon-pack` to be used with `org.wildfly:wildfly-galleon-pack` to provision an EE 8 server
```
Comment (jmesnil) [inline _posts/2022-04-20-WildFly-s2i-v2-Released.adoc:182]: Jakarta EE8

**Review by jmesnil — CHANGES_REQUESTED** [review]
@jfdenise That's great. I found some typos. Otherwise the content looks great, thanks

File: _posts/2022-04-14-WildFly-s2i-v2-Released.adoc [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+
+Our new cloud architecture is composed of 5 main components:
+
+* New WildFly S2I builder images 1.0.0 (link:https://quay.io/repository/wildfly/wildfly-s2i-jdk11[JDK11] and link:https://quay.io/repository/wildfly/wildfly-s2i-jdk17[JDK17])
+* New WildFly runtime images 1.0.0 (link:https://quay.io/repository/wildfly/wildfly-runtime-jdk11[JDK11] and link:https://quay.io/repository/wildfly/wildfly-runtime-jdk17[JDK17])
+* New link:https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin 3.0.0.Final] 
+* New link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[WildFly cloud Galleon feature-pack 1.0.0.Final] 
+* Evolved link:https://github.com/wildfly/wildfly-charts[WildFly Helm Charts]
+
+The new S2I workfow can be summerized as:
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply _posts/2022-04-14-WildFly-s2i-v2-Released.adoc]: Thanks fixed.

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+```
+helm install my-legacy-app -f helm.yaml wildfly/wildfly --version ^1.x
+```
+====
+
+== New WildFly S2I and runtime images
+
+We have released our new link:https://catalog.redhat.com/software/containers/ubi8/ubi-minimal/5c359a62bed8bd75a2c3fba8[UBI8 minimal] based images:
+
+* link:https://quay.io/repository/wildfly/wildfly-s2i-jdk11[https://quay.io/repository/wildfly/wildfly-s2i-jdk11] S2I builder image.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed.

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+
+* link:https://quay.io/repository/wildfly/wildfly-s2i-jdk17[https://quay.io/repository/wildfly/wildfly-s2i-jdk17] S2I builder image.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+
+* link:https://quay.io/repository/wildfly/wildfly-runtime-jdk11[https://quay.io/repository/wildfly/wildfly-runtime-jdk11] runtime image.
+
+* link:https://quay.io/repository/wildfly/wildfly-runtime-jdk17[https://quay.io/repository/wildfly/wildfly-runtime-jdk17] runtime image.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 7 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+* New link:https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin 3.0.0.Final] 
+* New link:https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[WildFly cloud Galleon feature-pack 1.0.0.Final] 
+* New link:https://github.com/wildfly/wildfly-charts[Helm Chart for WildFly 2.0] 
+
+The new S2I workfow can be summerized as:
+
+* Configure the WildFly Maven plugin to deploy your application in the WildFly server of your choice, finely tuned according to your needs (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/jsf-ejb-jpa/pom.xml[an example])!
+* Make your application sources available in a git repository (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/postgresql[an example]).
+* Define a Helm Chart for WildFly yaml file (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/postgresql/helm.yaml[an example]).
+* Install your helm file into OpenShift (link:https://github.com/wildfly/wildfly-s2i/blob/main/examples/jsf-ejb-jpa/README.md#example-steps[an example]). You are done!
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 2 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed.

**Review by jmesnil — APPROVED** [review]

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 3 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Yep, decoupled.

**Comment by jmesnil** [comment]
@jfdenise thanks

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 4 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed.

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 5 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 6 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: Fixed

File: _posts/2022-04-20-WildFly-s2i-v2-Released.adoc (hunk 8 above)
Comment (jfdenise) [reply _posts/2022-04-20-WildFly-s2i-v2-Released.adoc]: I clarified that an execution of the package goal is added.

**Pull request merged**